src/core/control/jobs/ExportTemplate.cpp

```diff
@@ -0,0 +1,193 @@
+#include "ExportTemplate.h"
+
+#include <cmath>  // for round
+#include <utility>
+
+#include "control/jobs/ProgressListener.h"  // for ProgressListener
+#include "model/Document.h"                 // for Document
+#include "model/XojPage.h"                  // for XojPage
+#include "util/i18n.h"                      // for _
+#include "view/DocumentView.h"              // for DocumentView
+
+ExportTemplate::ExportTemplate(Document* doc, fs::path filePath): doc{doc}, filePath{std::move(filePath)} {
+    ElementRangeVector fullRange;
+    fullRange.emplace_back(0, doc->getPageCount() - 1);
+    setExportRange(fullRange);
+}
+
+ExportTemplate::~ExportTemplate() {}
+
+void ExportTemplate::setExportBackground(const ExportBackgroundType exportBackground) {
+    this->exportBackground = exportBackground;
+}
+
+void ExportTemplate::setProgressListener(ProgressListener* progressListener) {
+    this->progressListener = progressListener;
+}
+
+void ExportTemplate::setExportRange(const PageRangeVector& exportRange) {
+    if (exportRange.empty()) {
+        ElementRangeVector fullRange;
+        fullRange.emplace_back(0, doc->getPageCount() - 1);
+        this->exportRange = fullRange;
+    } else {
+        this->exportRange = exportRange;
+    }
+}
+
+void ExportTemplate::setExportRange(const char* rangeStr) {
+    ElementRangeVector fullRange;
+    fullRange.emplace_back(0, doc->getPageCount() - 1);
+    exportRange = parseRange(rangeStr).value_or(fullRange);
```

Comment:
This looks like we could get into trouble with ranges of the form `2-`, with `std::numeric_limits<size_t>::max()` set as the maximal exported page index. (I haven't tested it yet)